src/common/components/repository-input/index.js

```diff
@@ -48,6 +50,24 @@ export class RepositoryInput extends Component {
     const isTouched = input.inputValue.length > 2;
     const isValid = !!input.repository && !input.error;
 
+    let submitButton;
+    if (conf.get('LP_API_USERNAME')) {
+      // Main path, for use in production.
+      submitButton = (
+        <Button type='submit' disabled={!isValid || input.isFetching}>
+          { input.isFetching ? 'Creating...' : 'Create' }
+        </Button>
+      );
+    } else {
+      // If the Launchpad API isn't configured, fall back to just verifying
+      // the repository.  This is handy in development.
+      submitButton = (
+        <Button type='submit' disabled={!isValid || input.isFetching}>
+          { input.isFetching ? 'Verifying...' : 'Verify' }
+        </Button>
+      );
+    }
```

Comment:
Breaking snippets of JSX out into separate methods generally results in cleaner code. There's a few examples [here](https://github.com/canonical-ols/build.snapcraft.io/pull/41/files#diff-b287bbebb1e8c31d3df3068963f0d3e9R55).